### ISSUE TITLE
[BUG dev] `type` key of serialized polymorphic belong_to relationships is underscored

### ISSUE THREAD

**Issue by cibernox** [Netflix/fast_jsonapi#318]
I have a business model where a `Board` has many `BoardStrip`s, and the board strip `belongs_to :strip, polymorphic: true`. 

Strip uses STI and has three subclasses.

When a board is serialized this is the shape of the payload:
```js
{
  "id": "4",
  "type": "board-strip",
  "attributes": {
    "sequence": 0
  },
  "relationships": {
    "board": {
      "data": {
        "id": "23",
        "type": "board"
      }
    },
    "strip": {
      "data": {
        "id": "15",
        "type": "work_strip"  // <------------------- here
      }
    }
  }
}
```

All my serializers have `set_key_transform :dash`, so I'd expect the type to be dasherized, like it is with non-polymorphic belongs_to relationships.

I am using the `dev` branch

**Comment by shishirmk** [comment]
Looks like this is an issue with how we have implemented polymorphic support. I am taking a look. I will update you when i have a fix ready.

**Comment by shishirmk** [comment]
The fix for this issue should be released in [Version 1.4](https://github.com/Netflix/fast_jsonapi/releases/tag/1.4)
